R/irace_helpers.R

```diff
@@ -0,0 +1,64 @@
+paradoxToIrace = function(ps){
+  assertClass(ps, "ParamSet")
+  # what about ParamUty = vector numeric/real
+  class_lookup = data.frame(paradox = c("ParamLgl","ParamInt","ParamDbl","ParamFct"),
+                            irace = c("c","i","r","c"), stringsAsFactors = FALSE)
+  name <- ps$ids()
+  type <- class_lookup[match(ps$class, class_lookup$paradox), 2]
+  range <- getIraceRange(ps)
+  if(ps$has_deps){
+    condition <- getIraceCondition(ps)
+  } else {
+    condition = NULL
+  }
+
+  # does mlr3tuning deal with trafo internally or passed to irace? irace only handle log
+  par.tab <- paste(name, '""', type, range, condition, collapse = "\n")
+  irace::readParameters(text = par.tab)
+}
+getIraceRange = function(ps){
+  rng = data.table(lower = ps$lower, upper = ps$upper, lvl = ps$levels)
+
+  apply(rng, 1, function(x){
+    if(is.na(x[[1]]))
+      return(paste0("(",paste0(x[[3]], collapse = ","),")"))
+    else
+      return(paste0("(",x[[1]],",",x[[2]],")"))
+  })
+}
+getIraceCondition = function(ps){
+  cond = apply(ps$deps, 1, function(x){
+    if(class(x[[3]])[1] == "CondEqual"){
+      condition = paste("|",x[[2]], "==", x[[3]]$rhs)
+    } else {
+      condition = paste("|",x[[2]],"%in%", paste0("c(",paste0(x[[3]]$rhs, collapse = ","),")"))
+    }
+  })
+
+  condition = data.table(id = ps$ids(), cond = "")
+  ind = match(ps$deps$id, condition$id)
+  condition$cond[ind] = cond
+  condition$cond
+}
+
+makeScenario = function(instance){
+  list(
+    targetRunner = targetRunner,
+    logFile = tempfile(),
+    instances = list(instance),
+    debugLevel = 0,
+    maxExperiments = ifelse(class(instance$terminator)[1] == "TerminatorEvals",
```

Comment:
don't use `ifelse` unless you want to have vectorization; use `if () x else y`

Reply:
also you are checking whether the terminator is based on number of evals, but then relying on the number given in the paramset. shouldn't you use the number given in the terminator?

Reply:
But that is a moot point because using the terminator in such a way is broken. We expect the user to create new `Terminator` classes, e.g. to combine other terminators; the terminators should be handled as black boxes imho. (Unfortunately I guess it's the best we can do for now so)

Reply:
> also you are checking whether the terminator is based on number of evals, but then relying on the number given in the paramset. shouldn't you use the number given in the terminator?

Sorry confused here, I am using the number given in the terminator... isn't that `instance$terminator$param_set$values$n_evals` ?

> But that is a moot point because using the terminator in such a way is broken. We expect the user to create new Terminator classes, e.g. to combine other terminators; the terminators should be handled as black boxes imho. (Unfortunately I guess it's the best we can do for now so)

Yes this is frustrating as this doesn't combine well with the package. The other problem is using different and confusing language. Terminator: "n_evals". irace: "n.instances" and "maxExperiments".

Reply:
> Sorry confused here, I am using the number given in the terminator... isn't that `instance$terminator$param_set$values$n_evals` ?

you are right, I misread that.